src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -60,15 +62,40 @@ public class RuntimeConfigLoaderJsonDeserializerTests
             new string[] { "@env(')", "@env()", "@env('envVarName')", "@env(''envVarName')", "@env('envVarName'')", "@env(''envVarName'')" },
             new string[] { "@env(')", "@env()", "envVarValue", "_envVarValue", "envVarValue_", "_envVarValue_" },
             false,
+            true,
             DisplayName = "Replacement strings with some matches.")]
-        public void CheckConfigEnvParsingTest(string[] repKeys, string[] repValues, bool exceptionThrown)
+        [DataRow(
+            new string[] { "@env('envVarName')", "@env(@env('envVarName'))", "@en@env('envVarName')", "@env'()@env'@env('envVarName')')')" },
+            new string[] { "envVarValue", "@env(envVarValue)", "@enenvVarValue", "@env'()@env'envVarValue')')" },
+            false,
+            false,
+            DisplayName = "Replacement strings that match, but shouldn't be replaced.")]
+        public void CheckConfigEnvParsingTest(
+            string[] repKeys,
+            string[] repValues,
+            bool exceptionThrown,
+            bool replaceEnvVar)
         {
             SetEnvVariables();
             try
             {
-                Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(GetModifiedJsonString(repValues), out RuntimeConfig expectedConfig), "Should read the expected config");
-                Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(GetModifiedJsonString(repKeys), out RuntimeConfig actualConfig), "Should read actual config");
+                RuntimeConfig expectedConfig;
+                if (replaceEnvVar)
+                {
+                    Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
+                        GetModifiedJsonString(repValues, @"""mssql"""), out expectedConfig, replaceEnvVar: replaceEnvVar),
+                        "Should read the expected config");
+                }
+                else
+                {
+                    Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(
+                        GetModifiedJsonString(repKeys, @"""@env('enumVarName')"""), out expectedConfig, replaceEnvVar: replaceEnvVar),
+                        "Should read the expected config");
```

Comment:
For Enum's i thought you were trying to always replace the env_var with an actual value otherwise deserializing the json would throw an exception?

Reply:
Thats right, fixed it.